cmd/manager/main.go

```diff
@@ -291,3 +310,435 @@ func serveCRMetrics(cfg *rest.Config) error {
+		log.Error(err, "")
+	}
+	operatorName, err := k8sutil.GetOperatorName()
+	if err != nil {
+		log.Error(err, "")
+	}
+	promethuesRule.ObjectMeta = metav1.ObjectMeta{
+		Namespace: operatorNamespace,
+		Name:      "mdc-operator-monitoring",
+		Labels:    labels,
+	}
+	job := fmt.Sprintf("%s-metrics", operatorName)
+	promethuesRule.Spec = monitoringv1.PrometheusRuleSpec{
+		Groups: []monitoringv1.RuleGroup{
+			{
+				Name: "general.rules",
+				Rules: []monitoringv1.Rule{
+					{
+						Alert: "MobileDeveloperConsoleOperatorContainerDown",
+						Expr: intstr.IntOrString{
+							Type:   intstr.String,
+							StrVal: fmt.Sprintf("absent(up{job=\"%s\"} == 1)", job),
+						},
+						For:         "5m",
+						Labels:      critical,
+						Annotations: mdcContainerDownAnnotations,
+					},
+				},
+			},
+		},
+	}
+}
+
+func reconcileGrafanaDashboard(grafanaDashboard *integreatlyv1alpha1.GrafanaDashboard) {
+	operatorNamespace, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		log.Error(err, "")
+	}
+	labels := map[string]string{
+		"monitoring-key": "middleware",
+		"prometheus":     "application-monitoring",
+	}
+	grafanaDashboard.ObjectMeta = metav1.ObjectMeta{
+		Namespace: operatorNamespace,
+		Name:      "mdc-operator-monitoring",
+		Labels:    labels,
+	}
+	grafanaDashboard.Spec = integreatlyv1alpha1.GrafanaDashboardSpec{
+		Name: "mobile-developer-console-operator.json",
+		Json: `{
+			"annotations": {
+				"list": [
+					{
+					"builtIn": 1,
+					"datasource": "-- Grafana --",
+					"enable": true,
+					"hide": true,
+					"iconColor": "rgba(0, 211, 255, 1)",
+					"name": "Annotations & Alerts",
+					"type": "dashboard"
+					}
+				]
+				},
+				"description": "Operator metrics",
+				"editable": true,
+				"gnetId": null,
+				"graphTooltip": 0,
+				"links": [],
+				"panels": [
+				{
+					"collapsed": false,
+					"gridPos": {
+					"h": 1,
+					"w": 24,
+					"x": 0,
+					"y": 0
+					},
+					"id": 9,
+					"panels": [],
+					"repeat": null,
+					"title": "Uptime",
+					"type": "row"
+				},
+				{
+					"aliasColors": {},
+					"bars": true,
+					"dashLength": 10,
+					"dashes": false,
+					"datasource": "Prometheus",
+					"fill": 1,
+					"gridPos": {
+					"h": 8,
+					"w": 24,
+					"x": 3,
+					"y": 1
+					},
+					"id": 1,
+					"legend": {
+					"avg": false,
+					"current": false,
+					"max": false,
+					"min": false,
+					"show": true,
+					"total": false,
+					"values": false
+					},
+					"lines": true,
+					"linewidth": 1,
+					"links": [
+					{
+						"type": "dashboard"
+					}
+					],
+					"nullPointMode": "null",
+					"percentage": true,
+					"pointradius": 5,
+					"points": false,
+					"renderer": "flot",
+					"seriesOverrides": [],
+					"spaceLength": 10,
+					"stack": false,
+					"steppedLine": false,
+					"targets": [
+					{
+						"expr": "kube_endpoint_address_available{namespace='` + operatorNamespace + `',endpoint='mobile-developer-console-operator'}",
```

Comment:
```suggestion
						"expr": "kube_endpoint_address_available{namespace='` + operatorNamespace + `',endpoint='mobile-developer-console-operator-metrics'}",
```

I _think_ this one is correct. I don't have cluster monitoring installed in the cluster where I'm verifying though, so I could be wrong.